File: Pyro.UI/src/app/models/current-user.ts
export class CurrentUser {
    public constructor(
        private readonly _accessToken: string,
        private readonly _expiresIn: Date,
        private readonly _id: string,
        private readonly _login: string,
        private readonly _roles: string[],
        private readonly _permissions: string[],
    ) {}

    public hasPermission(permission: string): boolean {
        return this._permissions.includes(permission);
    }

    public get accessToken(): string {
        return this._accessToken;
    }

    public get expiresIn(): Date {
        return this._expiresIn;
    }

    public get id(): string {
        return this._id;
    }

    public get login(): string {
        return this._login;
    }

    public get roles(): string[] {
        return this._roles;
    }

    public get permissions(): string[] {
        return this._permissions;
    }
}

File: Pyro.Api/Pyro.Domain.Shared/CurrentUserProvider/CurrentUser.cs
// Copyright (c) Dmytro Kyshchenko. All rights reserved.
// Licensed under the GPL-3.0 license. See LICENSE file in the project root for full license information.

namespace Pyro.Domain.Shared.CurrentUserProvider;

public record CurrentUser(
    Guid Id,
    string Login,
    IEnumerable<string> Roles,
    IEnumerable<string> Permissions)
{
    public bool HasPermission(string permission)
        => Permissions.Contains(permission);
}